File: solutions/4217-first-element-with-unique-frequency/first-element-with-unique-frequency.java
class Solution {
    public int firstUniqueFreq(int[] nums) {
        Map<Integer, Integer> counter = new HashMap<>();

        for (int num : nums) {
            counter.put(num, counter.getOrDefault(num, 0) + 1);
        }

        Map<Integer, Integer> freq = new HashMap<>();

        for (int current : counter.keySet()) {
            int count = counter.get(current);
            freq.put(count, freq.getOrDefault(count, 0) + 1);
        }

        for (int num : nums) {
            int count = counter.get(num);
            if (freq.get(count) == 1) return num;
        }

        return -1;
    }
}

File: solutions/4217-first-element-with-unique-frequency/first-element-with-unique-frequency.cpp
class Solution {
public:
    int firstUniqueFreq(vector<int>& nums) {
        unordered_map<int, int> mp,freq;
        for(int i = 0; i < nums.size(); i++) mp[nums[i]]++;
        for(auto& it: mp) freq[it.second]++;
        for(auto it: nums) if(freq[mp[it]] == 1) return it;
        return -1;
    }
};